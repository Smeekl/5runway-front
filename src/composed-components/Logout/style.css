.logout {
  padding: 60px 35px 0 35px;
}

.logout__title {
  font-weight: 700;
  font-size: 24px;
  color: var(--color-primary);
}

.logout__text {
  font-weight: 300;
  margin-bottom: 32px;
}

.logout__buttons {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.logout__button {
  width: 162px;
  height: 48px;
  border-radius: 15px;
  border: none;
  background: var(--color-primary-blue);
}

.logout__button > span {
  color: var(--color-secondary-white);
}

.logout__button:hover {
  background: var(--color-primary);
}

.logout__button:focus {
  background: var(--color-primary);
}
