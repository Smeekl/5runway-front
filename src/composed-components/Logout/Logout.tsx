import { useNavigate } from 'react-router-dom';
import Button from '@components/Button/Button';
import { useService } from '@services/services';
import { useAuth } from '@providers/auth.provider';
import './style.css';

const Logout = () => {
  const { userService } = useService();
  const navigate = useNavigate();
  const { logout } = useAuth();

  return (
    <div className="logout">
      <h3 className="logout__title">Logout</h3>
      <p className="logout__text">Do you really want to leave?</p>
      <div className="logout__buttons">
        <Button
          className="logout__button"
          onClick={() => {
            userService.logoutUser();
            logout();
            navigate('/');
          }}
        >
          Yes
        </Button>
        <Button className="logout__button">No</Button>
      </div>
    </div>
  );
};

export default Logout;
