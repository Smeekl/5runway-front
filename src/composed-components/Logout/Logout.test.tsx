import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import Logout from './Logout';

describe('Logout UI tests', () => {
  test('should render correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <Logout />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test('should be in the document', () => {
    const { container } = render(
      <MemoryRouter>
        <Logout />
      </MemoryRouter>
    );
    expect(container).toBeInTheDocument();
  });
});
