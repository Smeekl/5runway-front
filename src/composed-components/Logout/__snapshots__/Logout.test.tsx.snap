// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout UI tests should render correctly 1`] = `
<div>
  <div
    class="logout"
  >
    <h3
      class="logout__title"
    >
      Logout
    </h3>
    <p
      class="logout__text"
    >
      Do you really want to leave?
    </p>
    <div
      class="logout__buttons"
    >
      <button
        class="ant-btn ant-btn-default logout__button"
        type="button"
      >
        <span>
          Yes
        </span>
      </button>
      <button
        class="ant-btn ant-btn-default logout__button"
        type="button"
      >
        <span>
          No
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Main UI tests should render correctly 1`] = `
<div>
  <div
    class="logout"
  >
    <h3
      class="logout__title"
    >
      Logout
    </h3>
    <p
      class="logout__text"
    >
      Do you really want to leave?
    </p>
    <div
      class="logout__buttons"
    >
      <button
        class="ant-btn ant-btn-default logout__button"
        type="button"
      >
        <span>
          Yes
        </span>
      </button>
      <button
        class="ant-btn ant-btn-default logout__button"
        type="button"
      >
        <span>
          No
        </span>
      </button>
    </div>
  </div>
</div>
`;
